File: WinForms/CS/GridMainDemo/Modules/RowHeight.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using DevExpress.Utils;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Card;
using DevExpress.XtraEditors;
using DevExpress.Tutorials;

namespace DevExpress.XtraGrid.Demos {
    /// <summary>
    /// Summary description for RowHeight.
    /// </summary>
    public partial class RowHeight : TutorialControl {
        public RowHeight() {
            //
            // Required for Windows Form Designer support
            //
            CreateWaitDialog();
            InitializeComponent();
            InitNWindData();
            sbSwitching.Text = SwitchButtonCaption;
            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }
        protected override string[] WhatsThisCodeFileNames {
            get { return new string[] { "RowHeight" }; }
        }
        protected override string WhatsThisXMLFileName {
            get { return "RowHeight"; }
        }
        #region Init
        private void RowHeight_Load(object sender, System.EventArgs e) {
            InitAutoHeight();
        }
        public override DevExpress.XtraGrid.Views.Base.BaseView ExportView { get { return gridView1; } }
        private GridControl CurrentGrid { get { return gridControl1; } }
        protected override DataSet InitNWindXMLData(string dataFileName) {
            DataSet ds = new DataSet();
            SetWaitDialogCaption(Properties.Resources.LoadingTables);
            ds.ReadXml(dataFileName);
            
            CurrentGrid.DataSource = ds.Tables["Employees"];
            return ds;
        }
        private void InitAutoHeight() {
            if(CurrentGrid.MainView is GridView)
                cheAutoHeight.Checked = gridView1.OptionsView.RowAutoHeight;
            else
                cheAutoHeight.Checked = cardView1.OptionsBehavior.FieldAutoHeight;
            cheMemo.Checked = repositoryItemMemoEdit1.LinesCount == 0;
            chePicture.Checked = repositoryItemPictureEdit1.CustomHeight == 0;
            InitEnabledAutoHeight();
        }
        private void InitEnabledAutoHeight() {
            cheMemo.Enabled = chePicture.Enabled = cheAutoHeight.Checked;
        }
        #endregion
        #region Editing
        //<cheAutoHeight>
        private void chbAutoHeight_CheckedChanged(object sender, System.EventArgs e) {
            CheckEdit ce = sender as CheckEdit;
            if(CurrentGrid.MainView is GridView)
                gridView1.OptionsView.RowAutoHeight = ce.Checked;
            else
                cardView1.OptionsBehavior.FieldAutoHeight = ce.Checked;
            //<skip>
            InitEnabledAutoHeight();
            //</skip>
        }
        //</cheAutoHeight>
        //<cheMemo>
        private void chbMemo_CheckedChanged(object sender, System.EventArgs e) {
            CheckEdit ce = sender as CheckEdit;
            repositoryItemMemoEdit1.LinesCount = ce.Checked ? 0 : 1;
        }
        //</cheMemo>
        //<chePicture>
        private void chbPicture_CheckedChanged(object sender, System.EventArgs e) {
            CheckEdit ce = sender as CheckEdit;
            repositoryItemPictureEdit1.CustomHeight = ce.Checked ? 0 : 40;
            gridView1.TopRowIndex = 0;
        }
        //</chePicture>
        string ActiveViewCaption { get { return CurrentGrid.MainView == gridView1 ? Properties.Resources.GridView : Properties.Resources.CardView; } }
        string SwitchButtonCaption { get { return CurrentGrid.MainView == gridView1 ? Properties.Resources.ShowCardView : Properties.Resources.ShowGridView; } }

        //<sbSwitching>
        void sbSwitching_Click(object sender, System.EventArgs e) {
            //<skip>
            Cursor currentCursor = Cursor.Current;
            Cursor.Current = Cursors.WaitCursor;
            //</skip>
            if(CurrentGrid.MainView == cardView1)
                CurrentGrid.MainView = gridView1;
            else CurrentGrid.MainView = cardView1;
            //<skip>
            InitAutoHeight();
            CurrentGrid.Focus();
            CurrentGrid.MainView.LayoutChanged();
            Cursor.Current = currentCursor;
            RibbonMenuManager.AllowExport(CurrentGrid.MainView);
            sbSwitching.Text = SwitchButtonCaption;
            //</skip>
        }
        //</sbSwitching>
        #endregion
    }
}

File: WinForms/VB/GridMainDemo/Modules/RowHeight.vb
Imports System
Imports System.Drawing
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors

Namespace DevExpress.XtraGrid.Demos

    ''' <summary>
    ''' Summary description for RowHeight.
    ''' </summary>
    Public Partial Class RowHeight
        Inherits TutorialControl

        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            CreateWaitDialog()
            InitializeComponent()
            InitNWindData()
            sbSwitching.Text = SwitchButtonCaption
        '
        ' TODO: Add any constructor code after InitializeComponent call
        '
        End Sub

        Protected Overrides ReadOnly Property WhatsThisCodeFileNames As String()
            Get
                Return New String() {"RowHeight"}
            End Get
        End Property

        Protected Overrides ReadOnly Property WhatsThisXMLFileName As String
            Get
                Return "RowHeight"
            End Get
        End Property

#Region "Init"
        Private Sub RowHeight_Load(ByVal sender As Object, ByVal e As EventArgs)
            InitAutoHeight()
        End Sub

        Public Overrides ReadOnly Property ExportView As Views.Base.BaseView
            Get
                Return gridView1
            End Get
        End Property

        Private ReadOnly Property CurrentGrid As GridControl
            Get
                Return gridControl1
            End Get
        End Property

        Protected Overrides Function InitNWindXMLData(ByVal dataFileName As String) As DataSet
            Dim ds As DataSet = New DataSet()
            SetWaitDialogCaption(Properties.Resources.LoadingTables)
            ds.ReadXml(dataFileName)
            CurrentGrid.DataSource = ds.Tables("Employees")
            Return ds
        End Function

        Private Sub InitAutoHeight()
            If TypeOf CurrentGrid.MainView Is GridView Then
                cheAutoHeight.Checked = gridView1.OptionsView.RowAutoHeight
            Else
                cheAutoHeight.Checked = cardView1.OptionsBehavior.FieldAutoHeight
            End If

            cheMemo.Checked = repositoryItemMemoEdit1.LinesCount = 0
            chePicture.Checked = repositoryItemPictureEdit1.CustomHeight = 0
            InitEnabledAutoHeight()
        End Sub

        Private Sub InitEnabledAutoHeight()
            chePicture.Enabled = cheAutoHeight.Checked
            cheMemo.Enabled = chePicture.Enabled
        End Sub

#End Region
#Region "Editing"
        '<cheAutoHeight>
        Private Sub chbAutoHeight_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim ce As CheckEdit = TryCast(sender, CheckEdit)
            If TypeOf CurrentGrid.MainView Is GridView Then
                gridView1.OptionsView.RowAutoHeight = ce.Checked
            Else
                cardView1.OptionsBehavior.FieldAutoHeight = ce.Checked
            End If

            '<skip>
            InitEnabledAutoHeight()
        '</skip>
        End Sub

        '</cheAutoHeight>
        '<cheMemo>
        Private Sub chbMemo_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim ce As CheckEdit = TryCast(sender, CheckEdit)
            repositoryItemMemoEdit1.LinesCount = If(ce.Checked, 0, 1)
        End Sub

        '</cheMemo>
        '<chePicture>
        Private Sub chbPicture_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            Dim ce As CheckEdit = TryCast(sender, CheckEdit)
            repositoryItemPictureEdit1.CustomHeight = If(ce.Checked, 0, 40)
            gridView1.TopRowIndex = 0
        End Sub

        '</chePicture>
        Private ReadOnly Property ActiveViewCaption As String
            Get
                Return If(CurrentGrid.MainView Is gridView1, Properties.Resources.GridView, Properties.Resources.CardView)
            End Get
        End Property

        Private ReadOnly Property SwitchButtonCaption As String
            Get
                Return If(CurrentGrid.MainView Is gridView1, Properties.Resources.ShowCardView, Properties.Resources.ShowGridView)
            End Get
        End Property

        '<sbSwitching>
        Private Sub sbSwitching_Click(ByVal sender As Object, ByVal e As EventArgs)
            '<skip>
            Dim currentCursor As Cursor = Cursor.Current
            Cursor.Current = Cursors.WaitCursor
            '</skip>
            If CurrentGrid.MainView Is cardView1 Then
                CurrentGrid.MainView = gridView1
            Else
                CurrentGrid.MainView = cardView1
            End If

            '<skip>
            InitAutoHeight()
            CurrentGrid.Focus()
            CurrentGrid.MainView.LayoutChanged()
            Cursor.Current = currentCursor
            RibbonMenuManager.AllowExport(CurrentGrid.MainView)
            sbSwitching.Text = SwitchButtonCaption
        '</skip>
        End Sub
    '</sbSwitching>
#End Region
    End Class
End Namespace
